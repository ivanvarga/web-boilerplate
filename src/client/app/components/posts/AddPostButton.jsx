/* global document */

import React from 'react';

class AddPostButton extends React.Component {
  constructor(props) {
    super();
    this.handleClick = props.handleClick.bind(this);
  }
  render() {
    return (
      <button onClick={this.props.handleClick}> New post</button>
    );
  }
}

AddPostButton.propTypes = {
  handleClick: React.PropTypes.func.isRequired,
};

export default AddPostButton;
