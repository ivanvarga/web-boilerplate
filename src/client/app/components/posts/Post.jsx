/* global document */

import React from 'react';

function Post(props) {
  return (<div>{props.data}</div>);
}

Post.propTypes = {
  data: React.PropTypes.number,
};

export default Post;
