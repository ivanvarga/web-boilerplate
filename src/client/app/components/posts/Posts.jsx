/* global document */

import React from 'react';
import Post from './Post';

class Posts extends React.Component {
  render() {
    const posts = this.props.posts.map((post, i) => <Post data={post} key={i} />);
    return (
      <ul>
        {posts}
      </ul>
    );
  }
}

Posts.propTypes = {
  posts: React.PropTypes.arrayOf(React.PropTypes.number),
};

export default Posts;
