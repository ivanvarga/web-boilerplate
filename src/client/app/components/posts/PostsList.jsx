/* global document */

import React from 'react';

import Posts from './Posts';

import AddPostButton from './AddPostButton';

let i = 3;

class PostsList extends React.Component {
  constructor(props) {
    super();
    this.state = { posts: props.posts || [1, 2, 3] };
    this.handleClick = this.handleClick.bind(this);
  }
  getPostsCount() {
    return this.state.posts.length;
  }
  handleClick() {
    const posts = this.state.posts;
    i += 1;
    posts.push(i);
    this.setState({ posts });
  }
  render() {
    return (
      <div>
        <AddPostButton handleClick={this.handleClick} />
        <Posts posts={this.state.posts} />
      </div>
    );
  }
}

PostsList.propTypes = {
  posts: React.PropTypes.arrayOf(React.PropTypes.number),
};

export default PostsList;
