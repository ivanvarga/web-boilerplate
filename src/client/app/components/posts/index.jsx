import Posts from './Posts';
import Post from './Post';
import PostsList from './PostsList';
import AddPostButton from './AddPostButton';

export {
  Posts,
  Post,
  PostsList,
  AddPostButton,
};
