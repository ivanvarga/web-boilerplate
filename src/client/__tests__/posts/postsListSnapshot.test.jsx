/* global test expect mock*/

import renderer from 'react-test-renderer';

import React from 'react';

import { PostsList } from '../../app/components/posts/index';

test('PostsList renders correctly', () => {
  const list = renderer.create(
    <PostsList />,
  ).toJSON();
  expect(list).toMatchSnapshot();
});


