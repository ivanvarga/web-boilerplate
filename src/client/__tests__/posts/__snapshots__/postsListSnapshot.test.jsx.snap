exports[`test PostsList renders correctly 1`] = `
<div>
  <button
    onClick={[Function]}>
     New post
  </button>
  <ul>
    <div>
      1
    </div>
    <div>
      2
    </div>
    <div>
      3
    </div>
  </ul>
</div>
`;
